fn main() {
    // recoverable and unrecoverable errors
    //     ^              ^
    //     ^              ^
    // Result<T, E>    panic!


}
